mongo/integration/collection_test.go

```diff
@@ -824,95 +818,30 @@ func TestCollection(t *testing.T) {
 			got := x.Int32()
 			assert.Equal(mt, int32(1), got, "expected x value 1, got %v", got)
 
+			optionsDoc := bsoncore.NewDocumentBuilder().
```

Comment:
Just a note: using `NewDocumentBuilder` will make this an unclean backport because that functionality doesn't exist on the 1.4 branch. You can either keep this and manually change when backporting or change this to a `bson.M` and call `Marshal` to create a raw document. I don't really have a preference as this is test code, though.